src/Components/Nav/Nav.jsx

```diff
@@ -2,13 +2,52 @@ import React, { Component } from 'react';
 import './Nav.scss';
 
 class Nav extends Component {
+  listener = null;
+  constructor(props) {
+    super(props);
+    this.state = {
+      colorValue: ['announceBar red', 'announceBar green', 'announceBar blue'],
+      nav: false,
+      indexNum: 0,
+    };
+  }
+  componentDidMount() {
+    window.addEventListener('scroll', this.handleScroll);
+    setInterval(() => {
+      if (this.state.indexNum >= this.state.colorValue.length - 1) {
+        return this.setState({ indexNum: 0 });
+      }
+      this.setState(prevState => {
+        return {
+          indexNum: prevState.indexNum + 1,
+        };
+      });
+    }, 3000);
+  }
+  componentWillUnmount() {
+    window.removeEventListener('scroll');
+  }
+  handleScroll = () => {
+    if (window.pageYOffset > 40) {
+      if (!this.state.nav) {
+        this.setState({ nav: true });
+      }
+    } else {
+      if (this.state.nav) {
+        this.setState({ nav: false });
+      }
+    }
```

Comment:
이러면 좀 더 직관적일 것 같아요! 어떻게 생각하시나요?
그리고 비구조화 사용하면 좀 더 줄일 수 있습니다!
```suggestion
  const { nav } = this.state;
    if (window.pageYOffset > 40 && !nav) {
        this.setState({ nav: true });
        return;
    }
    if (window.pageYOffset <= 40 && nav) {
        this.setState({ nav: false });
        return;
      }
    }
```

Reply:
수정했습니다 관희님~~!!!😍